superbench/runner/runner.py

```diff
@@ -173,6 +177,94 @@ def fetch_results(self):    # pragma: no cover
+                    if results_file.parts[-3].endswith('_models'):
+                        benchmark_name = '{}/{}'.format(results_file.parts[-3], result['name'])
+                    if benchmark_name not in results_summary:
+                        results_summary[benchmark_name] = defaultdict(list)
+                    for metric in result['result']:
+                        metric_name = '{}/{}'.format(benchmark_name, metric)
+                        if metric_name not in reduce_ops:
+                            reduce_ops[metric_name] = result['reduce_op'][metric]
+                        elif reduce_ops[metric_name] != result['reduce_op'][metric]:
+                            logger.error('Inconsistent reduce type for metric: {}'.format(metric_name))
+                            continue
+
+                        results_summary[benchmark_name][metric].append(result['result'][metric])
+
+        results_summary = self.__merge_all_metrics(results_summary, reduce_ops)
+        with (node_path / 'results-summary.json').open(mode='w') as f:
+            json.dump(results_summary, f, indent=2)
+
+        return results_summary
+
+    def __merge_all_metrics(self, results_summary, reduce_ops):
+        """Merge metrics of all benchmarks in one node.
+
+        Args:
+            results_summary (dict): Summarized result of one node.
+            reduce_ops (dict): The reduce type of each metric.
+
+        Returns:
+            dict: Flattened result with metric as key.
+        """
+        metrics_summary = dict()
+        for benchmark_name in results_summary:
+            for metric in results_summary[benchmark_name]:
+                metric_name = '{}/{}'.format(benchmark_name, metric)
+                if metric_name not in reduce_ops:
+                    logger.error('Unknown reduce type for metric: {}'.format(metric_name))
+                    continue
+                if reduce_ops[metric_name] is not None:
+                    reduce_func = Reducer.get_reduce_func(ReduceType(reduce_ops[metric_name]))
+                    if reduce_func is None:
+                        logger.error(
+                            'Undefined reduce function {} for metric {}'.format(reduce_ops[metric_name], metric_name)
+                        )
+                        continue
+                    values = [reduce_func(list(result)) for result in zip(*results_summary[benchmark_name][metric])]
+                    for run_count in range(len(values)):
+                        metric_name = '{}/{}/{}'.format(benchmark_name, metric, str(run_count))
+                        metrics_summary[metric_name] = values[run_count]
+                else:
+                    for rank in range(len(results_summary[benchmark_name][metric])):
+                        for run_count in range(len(results_summary[benchmark_name][metric][rank])):
+                            metric_name = '{}/{}_{}/{}'.format(benchmark_name, metric, str(rank), str(run_count))
```

Comment:
same, maybe re-format to `name/run/metric/rank`, or `name:run/metric:rank`?
currently there's metric like "name_cublasCgemm_k_32_m_896_n_1792_transa_1_transb_0", it's not easy to parse whether there's rank in metric name.

@cp5555 to comment

Reply:
OK

Reply:
I think we can revise to `:`

Reply:
Done

Reply:
Have revised to `name/run/metric:rank`